ambry-clustermap/src/main/java/com.github.ambry.clustermap/CompositeClusterAgentsFactory.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import com.github.ambry.config.ClusterMapConfig;
+import java.io.IOException;
+import org.json.JSONException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A factory that creates a {@link CompositeClusterManager} and {@link ClusterParticipant}. Only one instance of each
+ * type of objects will ever be created by this factory.
+ */
+public class CompositeClusterAgentsFactory implements ClusterAgentsFactory {
+  private static final Logger logger = LoggerFactory.getLogger(CompositeClusterAgentsFactory.class);
+  private final StaticClusterAgentsFactory staticClusterAgentsFactory;
+  private final HelixClusterAgentsFactory helixClusterAgentsFactory;
```

Comment:
It does not strictly have to - but at this time, I don't see a reason to make these too generic. If we see value in keeping the composite cluster manager after transitioning to Helix, we can think of something.